src/main/java/org/opensearch/knn/index/Parameter.java

```diff
@@ -66,6 +66,35 @@ public T getDefaultValue() {
      */
     public abstract ValidationException validate(Object value);
 
+    /**
+     * Boolean method parameter
+     */
+    public static class BooleanParameter extends Parameter<Boolean> {
+        public BooleanParameter(String name, Boolean defaultValue, Predicate<Boolean> validator) {
+            super(name, defaultValue, validator);
+        }
+
+        @Override
+        public ValidationException validate(Object value) {
+            ValidationException validationException = null;
+            if (!(value instanceof Boolean)) {
+                validationException = new ValidationException();
+                validationException.addValidationError(
+                    String.format("Value not of type Boolean for Boolean " + "parameter \"%s\".", getName())
+                );
+                return validationException;
+            }
+
+            if (!validator.test((Boolean) value)) {
+                validationException = new ValidationException();
+                validationException.addValidationError(
+                    String.format("Parameter validation failed for Boolean " + "parameter \"%s\".", getName())
```

Comment:
nit: direct concat for strings is not needed. Also I think convention is to start error message from lower case, it will be wrapped into parent exception message anyway

Reply:
updated